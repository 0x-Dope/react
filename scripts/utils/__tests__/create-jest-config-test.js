const createConfig = require('../create-jest-config');

describe('create-jest-config', () => {

  it('creates a config object', () => {
    expect(createConfig({}, id => '<rootDir>/' + id)).toMatchSnapshot();
  });

  it('overwrites default options', () => {
    expect(createConfig(
      {
        testEnvironment: 'jest-environment-jsdom',
      },
      id => id
    ).testEnvironment).toBe('jest-environment-jsdom');
  });

  it('merges `setupFiles` and `moduleNameMapper` options', () => {
    expect(createConfig(
      {
        setupFiles: ['a'],
        moduleNameMapper: {
          b: 'c',
        },
      },
      id => id
    )).toMatchSnapshot();
  });

});
