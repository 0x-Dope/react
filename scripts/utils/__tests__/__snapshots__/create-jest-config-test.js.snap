exports[`create-jest-config creates a config object 1`] = `
Object {
  "automock": false,
  "moduleNameMapper": Object {
    "^[./a-zA-Z0-9$_-]+\\.(jpg|png|gif|eot|svg|ttf|woff|woff2|mp4|webm)$": "<rootDir>/config/jest/FileStub.js",
    "^[./a-zA-Z0-9$_-]+\\.css$": "<rootDir>/config/jest/CSSStub.js"
  },
  "persistModuleRegistryBetweenSpecs": true,
  "scriptPreprocessor": "<rootDir>/config/jest/transform.js",
  "setupFiles": Array [
    "<rootDir>/config/polyfills.js"
  ],
  "testEnvironment": "node"
}
`;

exports[`create-jest-config merges \`setupFiles\` and \`moduleNameMapper\` options 1`] = `
Object {
  "automock": false,
  "moduleNameMapper": Object {
    "^[./a-zA-Z0-9$_-]+\\.(jpg|png|gif|eot|svg|ttf|woff|woff2|mp4|webm)$": "config/jest/FileStub.js",
    "^[./a-zA-Z0-9$_-]+\\.css$": "config/jest/CSSStub.js",
    "b": "c"
  },
  "persistModuleRegistryBetweenSpecs": true,
  "scriptPreprocessor": "config/jest/transform.js",
  "setupFiles": Array [
    "config/polyfills.js",
    "a"
  ],
  "testEnvironment": "node"
}
`;
